Autopilot/Inc/SensorFusion.hpp

```diff
@@ -30,6 +28,58 @@ struct SFOutput_t {
     float longitudeSpeed; //m/s
 };
 
+#ifndef ATTITUDE_DATATYPES_HPP
+//Following structs store the raw sensor data so other modules can have direct access to them without including sensor header files
+struct IMU_Data_t
+{
+    float magx, magy, magz;
+    float accx, accy, accz;
+    float gyrx, gyry, gyrz; 
+
+    bool isDataNew; 
+    int sensorStatus; 
+    float utcTime; 
+};
+
+struct Airspeed_Data_t
+{
+    double airspeed;        
+
+    int sensorStatus;       
+    bool isDataNew;         
+    float utcTime;          
+};
+#endif
+
+struct Gps_Data_t
+{
+    long double latitude;  // 8 Bytes
+    long double longitude; // 8 Bytes
+    float utcTime;     // 4 Bytes. Time in seconds since 00:00 (midnight)
+    float groundSpeed; // in m/s
+    int altitude; // in m
+    short heading; // in degrees. Should be between 0-360 at all times, but using integer just in case
+    char numSatellites;    // 1 Byte
+	char fixStatus; //0 = No GPS, 1 = GPS fix, 2 = DGSP Fix, 3 = Estimated/Dead Recoking Fix
+
+    char sensorStatus; // 0 = no fix, 1 = gps fix, 2 = differential gps fix (DGPS) (other codes are possible)
+    bool dataIsNew; // true if data has been refreshed since the previous time GetResult was called, false otherwise.
+	bool timeIsValid;
+
+	//Added these so autopilot knows which data is new
+	bool ggaDataIsNew; //Position, altitude, time, and number of satellites
+	bool vtgDataIsNew; //Groundspeed and Heading
+};
```

Comment:
Indentation is a bit off. Can be fixed by selecting the lines, doing Shift+Tab, and then indenting them again!